src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -1139,22 +1139,48 @@ void Creature::SetLootRecipient(Unit* unit, bool withGroup)
         m_lootRecipient.Clear();
         m_lootRecipientGroup = 0;
         RemoveFlag(UNIT_DYNAMIC_FLAGS, UNIT_DYNFLAG_LOOTABLE | UNIT_DYNFLAG_TAPPED);
+        ResetAllowedLooters();
         return;
     }
 
-    //if (unit->GetTypeId() != TYPEID_PLAYER && !unit->IsVehicle())
-    //    return;
-
     Player* player = unit->GetCharmerOrOwnerPlayerOrPlayerItself();
     if (!player)                                             // normal creature, no player involved
         return;
 
     m_lootRecipient = player->GetGUID();
 
+    Map* map = GetMap();
+    if (map && map->IsDungeon() && (isWorldBoss() || IsDungeonBoss()))
+    {
+        AddAllowedLooter(m_lootRecipient);
+    }
+
     if (withGroup)
     {
         if (Group* group = player->GetGroup())
+        {
             m_lootRecipientGroup = group->GetGUID().GetCounter();
+
+            if (map && map->IsDungeon() && (isWorldBoss() || IsDungeonBoss()))
+            {
+                Map::PlayerList const& PlayerList = map->GetPlayers();
+                for (Map::PlayerList::const_iterator i = PlayerList.begin(); i != PlayerList.end(); ++i)
+                {
+                    if (Player* groupMember = i->GetSource())
+                    {
+                        if (groupMember->IsGameMaster() || groupMember->IsSpectator())
```

Comment:
Why the spectator check?

Reply:
Why don't we want GMs to be eligible for loot?

Reply:
Maybe it's time to add permission for GMs to loot items in worldserver.conf ?
Just in case if Administrator have any Gms as stuff

Reply:
In case of somebody uses the spectator addon - we don't want to include spectators as they are not participating in killing the bosses. The same is for GMs.

Reply:
> The same is for GMs.

It would be more correct to add a check if there are both GM and ordinary players in the group: loot according to the config is prohibited or allowed. To prevent abuse of GMs in favor of players

Reply:
@UltraNix You get a spectator flag when you enter an arena when you leave the arena it gets removed.

In a dungeon map a player can never be a spectator the IsSpectator is redundant here.
```c++
if (map && map->IsDungeon() && (isWorldBoss() || IsDungeonBoss()))
```

Reply:
I know there are some servers that use extended addon that enables raid spectating.

Reply:
Could you add a //@todo that adding RBAC permission here would be golden once it is implemented

Reply:
RBAC is completely useless here.

Reply:
Not really.

Reply:
What you mean?

Reply:
I mean, I think this could be a good place to use a RBAC perm as then you could choose if only some GMs should be allowed to participate in the loot and some not etc. I could see use for that in a production environment. However, as RBAC isn't implemented we just have to wait anyways